.teamcity/subprojects.json

```diff
@@ -671,6 +671,34 @@
     "functionalTests": true,
     "crossVersionTests": false
   },
+  {
+    "name": "plugins-groovy",
+    "path": "platforms/jvm/plugins-groovy",
+    "unitTests": true,
+    "functionalTests": true,
+    "crossVersionTests": false
+  },
+  {
+    "name": "plugins-java",
+    "path": "platforms/jvm/plugins-java",
+    "unitTests": true,
+    "functionalTests": true,
+    "crossVersionTests": false
+  },
+  {
+    "name": "plugins-jvm-test-suite",
+    "path": "platforms/jvm/plugins-jvm-test-suite",
+    "unitTests": false,
+    "functionalTests": true,
+    "crossVersionTests": false
+  },
+  {
+    "name": "plugins-jvm-test-suite-base",
```

Comment:
Is there a reason that the test suite project is split now? I think with the changes we made this is no longer a desired/necessary

Reply:
Test suite plugin -> Java plugin, Test suite base plugin.
Java plugin -> Test suite base plugin.

We could locate the `JvmTestSuite` and `JvmTestSuiteTarget` types right alongside the `java` plugin.  But they are used by the resolution consistency types, which is part of the extension, which is part of `java-base`.

So I think it's worth keeping them as a separate project to avoid further complications splitting `java` and `java-base` in the next round of work.

Reply:
What's the final picture going to look like here?

plugins-java depends on plugins-jvm-test-suite depends on plugin-java-base

plugins-jvm-test-suite-base would go away?

Reply:
Ideally, the final situation will be:

plugins-jvm-test-suite -> plugins-java -> plugins-java-base

Reply:
Even if we do this temporarily, I think these should be merged eventually.

The complication of implementing resolution consistency in the `JavaPluginExtension` makes this difficult. TBH this seems like another case of trying to split something that isn't ready to be split

Reply:
> plugins-jvm-test-suite -> plugins-java -> plugins-java-base

I guess it depends on what these projects have in them. Is `plugins-java` the current JavaPlugin? 

We should not make the jvm-test-suite depend on the JavaPlugin. It should only depend on "base", "java-base" or "jvm" level things.  The dependency goes the other way (JavaPlugin depends on jvm-test-suite).

Reply:
I know java automatically applies test-suites, I suppose I was thinking of test suites as something to be opted-into by java/groovy/kotlin, but this doesn't make much sense.

So plugins-java -> plugins-jvm-test-suite -> plugins-java-base as the goal then.  In that case should the test-suites and java(jvm)-base plugins also eventually be merged?

Reply:
> should the test-suites and java(jvm)-base plugins also eventually be merged

Are you referring to the merging of the plugin code or the projects?

I think we want to keep jvm-test-suite reusable by other JVM plugins. That's limited to just the built-in Groovy/Scala/Java ones right now. It's a little muddied with Kotlin since they're just applying `java` sometimes.

In theory, we should treat `jvm-test-suite` as the "add test functionality for the JVM ecosystem" plugin. As long as it doesn't require the Java plugin to be applied, that's OK for now.